pandas/core/reshape/merge.py

```diff
@@ -1200,7 +1209,34 @@ def _maybe_coerce_merge_keys(self):
                 typ = rk.categories.dtype if rk_is_cat else object
                 self.right = self.right.assign(**{name: self.right[name].astype(typ)})
 
+    def _create_cross_configuration(
+        self, _left, _right, on
+    ) -> Tuple["DataFrame", "DataFrame", str, str]:
+        if on is not None:
```

Comment:
this validation should be done first (IOW maybe move create_cross_configuration lower where you are calling)

Reply:
Just to be sure, that I understand you correctly: Calling the function ``_create_cross_configuration`` should be done after calling ``_validate_specification``? in this case I have moved the parts around